tests/test_scripts/__snapshots__/gengraphql/meta.graphql

```diff
@@ -1,4 +1,43 @@
 # metamodel_version: 1.7.0
+enum CordialnessEnum
+  {
+    heartfelt
+    hateful
+    indifferent
+  }
+
+enum DiagnosisType
+  {
+    TODO
+  }
+
+enum EmploymentEventType
+  {
+    HIRE
+    FIRE
+    PROMOTION
+    TRANSFER
+  }
+
+enum FamilialRelationshipType
+  {
+    SIBLING_OF
+    PARENT_OF
+    CHILD_OF
+  }
+
+enum LifeStatusEnum
+  {
+    LIVING
+    DEAD
+    UNKNOWN
+  }
+
+enum other codes
```

Comment:
this is invalid graphql, the name needs to be CamelCased.

Reply:
(and if tests pass with invalid graphql, that means the tests need to be improved - it would probably be a good idea to add a dep for some graphql python library that can do schema validation rather than roll our own here, bc otherwise we would/should end up with a bunch of unit tests like `def test_gql_enum_names()` where we have to specifically test for camelcasing and etc.)

Reply:
test_compliance will eventually take care of all this. Each feature against each framework (where each framework declaratively specifies the profile of linkml that is supported). This is working well for json-schema, and anything that is a plugin for the validator framework can easily be slotted in.

We don't have anything like that for graphql yet, it's not clear the best way to use graphql as a data validator
https://stackoverflow.com/questions/45964170/how-to-use-a-graphql-schema-for-json-schema-like-data-validation

But we can at least do graphsql-schema level checks

But let's take some of this discussion here:
https://github.com/orgs/linkml/discussions/1549

Reply:
Graphql does not enforce snake-/camel-/whatever-casing. The rules are: only letters, digits and underscores and no heading digit. Otherwise total freedom. There are some best-practices though. Not enforcing them here to keep LinkML <-> GraphQL names as far as possible.